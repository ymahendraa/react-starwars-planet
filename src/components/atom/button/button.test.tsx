import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import StyledButton from '.';

describe('StyledButton Component', () => {
    it('renders button with default styles', () => {
        const { getByRole } = render(<StyledButton>Click me</StyledButton>);
        const button = getByRole('button');

        expect(button).toMatchSnapshot();
        expect(button).toHaveStyleRule('background', '-webkit-linear-gradient(#ff00e6, #02c4ff)');
        expect(button).toHaveStyleRule('color', 'white');
        expect(button).toHaveStyleRule('padding', '10px 20px');
        expect(button).toHaveStyleRule('border-radius', '10px');
        expect(button).toHaveStyleRule('border', 'none');
        expect(button).toHaveStyleRule('cursor', 'pointer');
    });

    it('applies custom styles', () => {
        const { getByRole } = render(
            <StyledButton
                background="blue"
                hoverbackground="blue"
                color="yellow"
                borderradius="5px"
            >
                Click me
            </StyledButton>
        );
        const button = getByRole('button');

        expect(button).toMatchSnapshot();
        expect(button).toHaveStyleRule('background', 'blue');
        expect(button).toHaveStyleRule('color', 'yellow');
        expect(button).toHaveStyleRule('border-radius', '5px');
    });

    it('applies disabled styles', () => {
        const { getByRole } = render(<StyledButton disabled>Click me</StyledButton>);
        const button = getByRole('button');

        expect(button).toMatchSnapshot();
        expect(button).toHaveStyle('background: #d3d3d3');
        expect(button).toHaveStyle('cursor: not-allowed');

    });

    it('handles click event', () => {
        const handleClick = jest.fn();
        const { getByRole } = render(<StyledButton onClick={handleClick}>Click me</StyledButton>);
        const button = getByRole('button');

        fireEvent.click(button);

        expect(handleClick).toHaveBeenCalledTimes(1);
    });
});
