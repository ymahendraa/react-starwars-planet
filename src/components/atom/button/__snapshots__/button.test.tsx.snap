// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton Component applies custom styles 1`] = `
.c0 {
  background: blue;
  color: yellow;
  border: none;
  font-weight: bold;
  padding: 10px 20px;
  border-radius: 5px;
  border: none;
  cursor: pointer;
}

.c0:hover {
  background: blue;
}

.c0:disabled {
  background: #d3d3d3;
  cursor: not-allowed;
}

<button
  background="blue"
  borderradius="5px"
  class="c0"
  color="yellow"
  hoverbackground="blue"
>
  Click me
</button>
`;

exports[`StyledButton Component applies disabled styles 1`] = `
.c0 {
  background: -webkit-linear-gradient(#ff00e6, #02c4ff);
  color: white;
  border: none;
  font-weight: bold;
  padding: 10px 20px;
  border-radius: 10px;
  border: none;
  cursor: pointer;
}

.c0:hover {
  background: -webkit-linear-gradient(#67065d, #085871);
}

.c0:disabled {
  background: #d3d3d3;
  cursor: not-allowed;
}

<button
  class="c0"
  disabled=""
>
  Click me
</button>
`;

exports[`StyledButton Component renders button with default styles 1`] = `
.c0 {
  background: -webkit-linear-gradient(#ff00e6, #02c4ff);
  color: white;
  border: none;
  font-weight: bold;
  padding: 10px 20px;
  border-radius: 10px;
  border: none;
  cursor: pointer;
}

.c0:hover {
  background: -webkit-linear-gradient(#67065d, #085871);
}

.c0:disabled {
  background: #d3d3d3;
  cursor: not-allowed;
}

<button
  class="c0"
>
  Click me
</button>
`;
